Removing dependencies to speed up startup

hubertp opened the pull request:
### Pull Request Description

Reducing the number of dependencies. Explicit `cats` are almost gone (present in `cli`). `enumeration` is completely gone.  `cats` is also still included implicitly via `io.circe` but that's a different kind of beast.
Also, really removed `jackson` from dependencies by fixing the dependency on `http-test-helper`.

### Important Notes

In a number of places importing all cats implicits could be simply replaced with a single or two method calls. Not to mention that this will reduce compilation times due to reduced implicit search space.

One example of how the changes affect performance (not only startup):

Before:
![Screenshot from 2024-06-11 12-05-24](https://github.com/enso-org/enso/assets/292128/a1a772a9-635d-4a16-a543-e2fd2124a22c)
Now: 
![Screenshot from 2024-06-11 14-27-47](https://github.com/enso-org/enso/assets/292128/b17c7fcc-9a6d-48b9-8200-60708354ee03)
(frequently executed)

![Screenshot from 2024-06-12 12-46-34](https://github.com/enso-org/enso/assets/292128/31bc4dfd-4edc-45c9-9c5d-13e3472089b9)
Also appears to be gone.

This PR is by no means finished. The purge will continue in follow up PRs.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [ ] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- [ ] Unit tests have been written where possible.

hubertp committed:
Removing cats and other animals

Reducing the number of dependencies. In a number of places importing all
cats implicits could be simply replaced with a single or two method
calls. Not to mention that this will reduce compilation times.

hubertp committed:
nit

hubertp commented:
I liked the other PR title more but it wasn't descriptive enough.

JaroslavTulach reviewed (approved):
Removing cats is good.

hubertp committed:
Remove type-level enumeration lib

No need for some fancy type level enumerations. Plain, builtin
enumerations are working just fine.

hubertp committed:
One more cat is gone, jackson no longer in jars

`http-test-helper` was being added as a runtime dependency, meaning that
jackson was still includede in `runtime.jar` and `runner.jar`.
Rewritten one more cat's traverse to avoid implicits.

hubertp committed:
nits

hubertp committed:
licensing

hubertp committed:
Add jackson test

JaroslavTulach committed:
Javadoc without errors
